### PR TITLE
Defer evaluation of strings

### PR DESCRIPTION
**Pull request by dyve**
Fixes: #8 

Defer evaluation of strings to improve i18n compatibility.

### CODE REVIEW

**Comment by dyve** [comment]
Dank voor je geduld met #8 :-)

File: tests/test_icon.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    def test_laziness(self):
+        self.assertEqual(
+            icon("user", title=_("user"), renderer=BaseRenderer),
+            '<i class="user" title="user"></i>',
+        )
+        with translation.override("nl"):
+            self.assertEqual(
+                icon("user", title=_("user"), renderer=BaseRenderer),
+                '<i class="user" title="gebruiker"></i>',
+            )
```
Comment (jieter) [inline tests/test_icon.py:21]: This test does not proof lazyness, I'd expect something like this to work:

```suggestion
        user_icon = icon("user", title=_("user"), renderer=BaseRenderer)
        with translation.override("nl"):
            self.assertEqual(user_icon, '<i class="user" title="gebruiker"></i>')
```

**Comment by dyve** [comment]
Dank voor de uitleg. Nieuwe poging.

**Review by jieter — APPROVED** [review]
Looks good to me!

**Pull request merged**